Source/AwsCommonRuntimeKit/io/retryer/CRTAWSRetryToken.swift

```diff
@@ -11,7 +11,4 @@ public final class CRTAWSRetryToken {
         self.rawValue = rawValue
     }
 
-    deinit {
-        aws_retry_token_release(rawValue)
-    }
```

Comment:
I don't know enough about Swift/C bindings and object lifetime to say whether this is the right call.

At first glance, the pre-existing deinit here looks right. A "wrapper" swift class ensure the C object stays alive by keeping 1 reference-count on the C object. Then releases the reference-count when the Swift object is destroyed. If this was the last reference-count on the C object, it will be cleaned up. If something else has acquired a reference to the C object, then the C object stays alive until that reference is also realeased.

But there is a double free going on. And I can see that CRTAWSRetryStrategy ALSO has a `releaseToken()` function that will ALSO release a reference-count on the C object.

In Swift, someone should "release" the token by ceasing to reference it, not by manually calling a release() function, right?

But I don't know Swift well enough to understand the nitty gritty of what's going on in this code

Reply:
Also, it would be great to add a unit test that reproduces this crash. Then we could be sure that the fix actually fixes it. The unit test could just use these retry classes directly, it wouldn't actually need to do any HTTP calls or anything

Reply:
Revised the PR with PRs also prepared for the other packages.

https://github.com/awslabs/aws-sdk-swift/pull/592
https://github.com/awslabs/smithy-swift/pull/422